nservicebus/ravendb/unit-of-work.md

```diff
@@ -14,31 +14,55 @@ Share the session between the message handler(s) and the actual unit of work imp
 
 Instead, beginning with NServiceBus version 3, use the new [support for child containers](/nservicebus/containers/child-containers.md). This means that all dependencies configured as a single call effectively become static within the context of one transport message.
 
-To resolve a RavenDB document session from the container, add the following configuration (StructureMap is used but any of the other containers except Spring and Unity would work):
-
+To resolve a RavenDB document session from the container, add the following configuration (StructureMap is used but any of the other containers except Spring and Unity would work):  
 
 ```C#
 var store = new DocumentStore {Url = "http://localhost:8080", DefaultDatabase = "MyDatabase"};
 store.Initialize();
  
-ObjectFactory.Configure(c =>
-    {
-        c.For<IDocumentStore>()
-            .Singleton()
-            .Use(store);
-        c.For<IDocumentSession>()
-            .Use(ctx => ctx.GetInstance<IDocumentStore>()
-                        .OpenSession());
-        c.For<IManageUnitsOfWork>()
-            .Use<RavenUnitOfWork>();
-    });
- 
-Configure.With()
-            .StructureMapBuilder(ObjectFactory.Container);
+//ObjectFactory.Configure(c =>  // objectFactory has been deprecated in structureMap
+Container container = new Container( c => 
+                    {
+                        c.For<IDocumentStore>().Singleton()
+                            .Use(dbStore);
+                        c.For<IDocumentSession>()
+                            .Use(ctx => ctx.GetInstance<IDocumentStore>()
+                                        .OpenSession());
+                        c.For<IManageUnitsOfWork>()
+                            .Use<RavenUnitOfWork>();
+                    });
+configuration.UseContainer<StructureMapBuilder>(c =>
+                       c.ExistingContainer(container));
+
+// old version: 
+// Configure.With()
+//            .StructureMapBuilder(ObjectFactory.Container);
+                       
+// rest of contiguration goes here...
+// configuration.UseSerialization<JsonSerializer>(); // default is XmlSerializer...
+configuration.UsePersistence<RavenDBPersistence>();
+
+
 ```
 
 The above code tells the container to create a new `IDocumentSession` using the specified lambda. The fact that all message processing is done using a child container means that all message handlers processing the message get the same session instance.
 
+If using the NServiceBus host via Nuget, you can add the above code inside the Customize() method in the EnpointConfig. In the case of StructureMap, you'll need get NServiceBus.StructureMap through Nuget, and add at the top of the class:
```

Comment:
Do we really need this, the tooling would just prompt the user and add the usings?